lib/ruby_smb/server.rb

```diff
@@ -0,0 +1,39 @@
+require 'socket'
+
+module RubySMB
+  # This class provides the SMB server core. Settings that are relevant server wide are managed by this object.
+  # Currently, the server only supports negotiating and authenticating requests. No other server functionality is
+  # available at this time. The negotiating and authentication is supported for SMB versions 1 through 3.1.1.
+  class Server
+    require 'ruby_smb/server/server_client'
+    require 'ruby_smb/gss/provider/ntlm'
+
+    Connection = Struct.new(:client, :thread)
+
+    # @param server_sock the socket on which the server should listen
+    # @param [Gss::Provider] the authentication provider
+    def initialize(server_sock: nil, gss_provider: nil)
+      server_sock = ::TCPServer.new(445) if server_sock.nil?
+
+      @server_guid = Random.new.bytes(16)
```

Comment:
Why is this `Random.new.bytes` when the NTLM generates challenges with  `SecureRandom.bytes`?

Reply:
The Server GUID isn't sensitive. It can be sent to unauthenticated entities so it didn't seem worthwhile to take it from the potentially limited source of secure entropy.